### PR TITLE
Tta #266 add create cohort

### PR DESCRIPTION
**Pull request by TomBrockington**
Have added radio but for student or teacher role selection on sign up

### CODE REVIEW

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -5,7 +5,19 @@ import { Stack } from '@mui/material';
...
 
-const Header = ({ companyName }) => {
+
+const Header = ({ companyName}) => {
+  
+
+ 
+ 
+
+ /*function createCohort(event){
```
Comment (Auenc) [inline src/components/Header/Header.jsx:15]: Lets remove this commented code

File: src/components/posts/PostsPage.js [hunk 2]
```diff
@@ -1,32 +1,62 @@
-import { useState, useEffect } from 'react';
-import { useNavigate } from 'react-router-dom';
-import PostForm from './PostForm';
-import client from '../../utils/client';
-import './style.css';
+import { useState, useEffect } from "react";
```
Comment (Auenc) [inline src/components/posts/PostsPage.js:1]: Let's make sure we're using single quotes in the app

File: src/components/posts/PostsPage.js [hunk 3]
```diff
@@ -1,32 +1,62 @@
...
 
 const PostsPage = () => {
-  const [post, setPost] = useState({ content: '' });
-  const [postResponse, setPostResponse] = useState('');
+  const [post, setPost] = useState({ content: "" });
+  const [postResponse, setPostResponse] = useState("");
   const [posts, setPosts] = useState([]);
+  const [isTeacher, setIsTeacher] = useState(false);
+  const [cohorts, setCohorts] = useState([]);
+  // const [createResponse, setCreateResponse] = useState("");
```
Comment (Auenc) [inline src/components/posts/PostsPage.js:18]: lets remove this commented code

File: src/components/posts/PostsPage.js [hunk 4]
```diff
@@ -1,32 +1,62 @@
...
+
+  const tokenKey = process.env.REACT_APP_USER_TOKEN;
+
+
+  //useEffect(() => {
```
Comment (Auenc) [inline src/components/posts/PostsPage.js:23]: and this commented code

File: src/components/posts/PostsPage.js [hunk 5]
```diff
@@ -1,32 +1,62 @@
...
+  //   client.get("/cohort").then((res) => setCohorts(res.data.data.cohort));
+  // }, []);
+
   let navigate = useNavigate();
 
   useEffect(() => {
-    client.get('/posts').then((res) => setPosts(res.data.data.posts));
+    const token = localStorage.getItem(tokenKey);
+    const decoded = jwt_decode(token);
+    console.log(decoded);
```
Comment (Auenc) [inline src/components/posts/PostsPage.js:32]: Let's remove all of these console.logs to keep the console clear.

File: src/components/users/login/LoginPage.js [hunk 6]
```diff
@@ -1,28 +1,39 @@
-import { Link } from 'react-router-dom';
-import { useState, useEffect } from 'react';
-import UserForm from './UserForm';
-import userBlankData from '../utils/userHelpers';
-import client from '../../../utils/client';
-import { useNavigate } from 'react-router-dom';
+import { Link } from "react-router-dom";
```
Comment (Auenc) [inline src/components/users/login/LoginPage.js:1]: Let's change these double quotes to single quotes

File: src/components/users/login/LoginPage.js [hunk 7]
```diff
@@ -1,28 +1,39 @@
...
     client
-      .post('/login', user)
+      .post("/login", user)
       .then((res) => {
-        localStorage.setItem(process.env.REACT_APP_USER_TOKEN, res.data.data.token);
+        localStorage.setItem(
+          process.env.REACT_APP_USER_TOKEN,
+          res.data.data.token
+        );
+        console.log(res.data.data.user.role);
```
Comment (Auenc) [inline src/components/users/login/LoginPage.js:30]: Let's get rid of the console.logs

File: src/components/users/registration/RegistrationPage.js [hunk 8]
```diff
@@ -1,39 +1,46 @@
 import { Link } from "react-router-dom";
 import { useState } from "react";
-import UserForm from './UserForm'
-import userBlankData from '../utils/userHelpers'
-import client from '../../../utils/client'
-import './style.css'
+import UserForm from "./UserForm";
```
Comment (Auenc) [inline src/components/users/registration/RegistrationPage.js:3]: Double quotes to single quotes

**Review by Auenc — CHANGES_REQUESTED** [review]
Looks good to me, other than you have a lot of double quotes instead of single quotes, as well as a lot of console.logs that don't need to be there.

Remove these and the PR should be good to go!

File: src/components/posts/PostsPage.js (hunk 3 above)
Comment (deliailu27) [reply src/components/posts/PostsPage.js]: deleted

File: src/components/posts/PostForm.js [hunk 9]
```diff
@@ -1,14 +1,22 @@
...
-const PostForm = ({handleSubmit, handleChange}) => {
-
-  return(
+const PostForm = ({ handleSubmit, handleChange }) => {
+  return (
     <form className="post-form" onSubmit={handleSubmit}>
-      <TextField className="user-form-input" type="text" label="New Post" variant="outlined" name="content" onChange={handleChange} />
-      <Button type="submit" variant="contained">Submit</Button>
+      <TextField
+        className="user-form-input"
```
Comment (Auenc) [inline src/components/posts/PostForm.js:8]: Some single quotes here

File: src/components/posts/PostsPage.js [hunk 10]
```diff
@@ -27,6 +48,7 @@ const PostsPage = () => {
   };
 
   const handleChange = (event) => {
+    console.log('event', event.target);
```
Comment (Auenc) [inline src/components/posts/PostsPage.js:51]: console.log to be removed here

File: src/components/posts/PostsPage.js [hunk 11]
```diff
@@ -41,18 +63,53 @@ const PostsPage = () => {
     navigate('../', { replace: true });
   };
 
+  function createCohort(event) {
+    event.preventDefault();
+    client.post('/cohort').then((res) => {
+      console.log('create cohort res data:', res.data);
```
Comment (Auenc) [inline src/components/posts/PostsPage.js:69]: console.log

File: src/components/posts/PostsPage.js [hunk 12]
```diff
@@ -41,18 +63,53 @@ const PostsPage = () => {
...
-      <section className='posts-section'>
-        <button id='user-signout-button' onClick={signOut}>
+      <Header
+        companyName={`Cohort Manager 2.0`}
+        isTeacher={isTeacher}
+        createCohort={createCohort}
+      />
+
+      {isTeacher && (
+        <div className="teacher-section">
```
Comment (Auenc) [inline src/components/posts/PostsPage.js:87]: double quotes

File: src/components/profile/Profile.js [hunk 13]
```diff
@@ -1,35 +1,34 @@
...
-        </>
-    )
-}
+  const { first_name, last_name, biography, github_url } = profileData;
+  return (
+    <>
+      <Header companyName={`Cohort Manager 2.0`} />
+      <div className="profile">
+        <Avatar
+          alt="Profile Pic"
```
Comment (Auenc) [inline src/components/profile/Profile.js:15]: double quotes here

File: src/components/users/login/LoginPage.js [hunk 14]
```diff
@@ -38,14 +48,15 @@ const LoginPage = () => {
   };
 
   return (
-    <div className='login-page'>
+    <div className="login-page">
```
Comment (Auenc) [inline src/components/users/login/LoginPage.js:51]: double quotes

File: src/components/users/registration/RegistrationPage.js [hunk 15]
```diff
@@ -1,39 +1,45 @@
...
       ...user,
       [name]: value,
     });
-  }
+  };
 
   return (
     <div className="registration-page">
-      <Link id="user-registration-link" to="/signup">sign up!</Link> <Link id="user-login-link" to="/">login!</Link>
+      <Link id="user-registration-link" to="/signup">
```
Comment (Auenc) [inline src/components/users/registration/RegistrationPage.js:32]: double quotes

File: src/components/users/registration/UserForm.js [hunk 16]
```diff
@@ -1,19 +1,79 @@
...
-      <Button id="user-submit-button" type="submit" variant="contained">Submit</Button>
+      <TextField
+        className="user-form-input"
+        label="First Name"
+        variant="outlined"
+        name="first_name"
+        onChange={handleChange}
+      />
+      <TextField
+        className="user-form-input"
```
Comment (Auenc) [inline src/components/users/registration/UserForm.js:22]: double quotes

File: src/constants.js [hunk 17]
```diff
@@ -7,7 +7,8 @@ const constants = {
       "email",
       "password",
       "biography",
-      "github_url"
+      "github_url",
```
Comment (Auenc) [inline src/constants.js:10]: double quotes

**Review by Auenc — CHANGES_REQUESTED** [review]
There are still quite a few double quotes in some of these files, and one or two console.logs

**Review by Auenc — DISMISSED** [review]
Looks good to me

**Review by Auenc — DISMISSED** [review]
Looks good to me!

**Review by deliailu27 — APPROVED** [review]

**Review by Auenc — DISMISSED** [review]
Looks good to me

**Review by deliailu27 — APPROVED** [review]

**Review by deliailu27 — APPROVED** [review]

File: cypress/integration/login-logout/login-logout.spec.js [hunk 18]
```diff
@@ -1,57 +1,18 @@
...
+describe("User Login Logout", () => {
   describe("a valid user", () => {
     beforeEach(() => {
-      cy.intercept('POST', 'http://localhost:4000/user', { fixture: 'registration/valid-user.json' })
-      cy.intercept('POST', 'http://localhost:4000/login', { fixture: 'login-logout/valid-user.json' })
-      cy.intercept('GET', 'http://localhost:4000/posts', { fixture: 'posts/valid-posts.json' })
-      cy.visit('/')
-    })
-
-    it('can login a user and be redirected to the posts page', () => {
```
Comment (vherus) [inline cypress/integration/login-logout/login-logout.spec.js:10]: Why is this test being removed?

File: src/components/posts/PostsPage.js [hunk 19]
```diff
@@ -44,12 +71,43 @@ const PostsPage = () => {
     navigate("../", { replace: true });
   };
 
+  function createCohort(event) {
+    event.preventDefault();
+    client.post('/cohort').then((res) => {
+     
+      if (res.data.status === 'success') {
+        alert(`cohort ${res.data.data.cohort.id} created`);
```
Comment (vherus) [inline src/components/posts/PostsPage.js:79]: Let's not use `alert`, go with conditionally rendering an element to display a message instead. Browsers can block alerts.

File: src/components/posts/PostsPage.js [hunk 20]
```diff
@@ -44,12 +71,43 @@ const PostsPage = () => {
...
+      } else {
+        alert(`error`);
```
Comment (vherus) [inline src/components/posts/PostsPage.js:81]: Same as above

**Review by vherus — CHANGES_REQUESTED** [review]

File: src/components/posts/PostsPage.js (hunk 20 above)
Comment (deliailu27) [reply src/components/posts/PostsPage.js]: added catch

**Review by Auenc — APPROVED** [review]
Looks good to me!

**Review by deliailu27 — APPROVED** [review]

**Pull request merged**